src/app/components/admin/admin.component.ts

```diff
@@ -13,25 +15,47 @@ import { Router } from '@angular/router';
 
 export class AdminComponent implements OnInit {
   /**
-   * @constructor 
-   * @param router Provides an instance of a router 
+   * @constructor
+   * @param router Provides an instance of a router
    */
-  constructor(private router: Router) { }
+  constructor(private router: Router, private adminservice: UserService) { }
 
-  /**
-   * This is an OnInit function that is evoked once.
-   */
+   listofUsers: User[];
+   allListofUsers: User[];
 
+   truthy: string = 'btn btn-success';
+   falsy: string = 'btn btn-danger';
+   searchText;
   ngOnInit() {
-  }
+
+    this.adminservice.showAllUser()
+    .subscribe(
+      data=> {
+        this.listofUsers = data;
+        this.allListofUsers = data;
+      }
+    )
+}
   /**
-   * Function that takes no parameters. 
+   * Function that takes no parameters.
    * It will clear the sesssion storage.
-   * @return {void} 
-   * 
+   * @return {void}
+   *
    */
   logout() {
     sessionStorage.clear();
     this.router.navigate(['']);
   }
+
+  searchUser(){
+    this.listofUsers = this.allListofUsers.filter(item => item.userName.toLowerCase().includes(this.searchText.toLowerCase()));
+  }
+
+  banning(obj: User, userid: number, userName: string, firstName: string, lastName: string, email: string, phoneNumber:string, batch: object, active: boolean) {
+    console.log(obj);
```

Comment:
@jasonyang1992 
- remove console log here 
- is there a reason for passing all of these parameters in along with the user object? wouldn't all of those fields already be a part of the user object?